Add Job Guard utility (#904)
kyma-project/test-infra: development/tools/cmd/jobguard/README.md

# Job Guard 

## Overview

Job Guard is a simple tool that fetches all statuses for GitHub pull requests and waits for some to finish.

> The main purpose of the Job Guard is to delay running integration jobs, which depend on components ones. The utility is run as a guard for integration tests.

Reviewer: ```suggestion
The main purpose of the Job Guard is to delay running integration jobs that depend on components jobs. This tool is run as a guard for integration tests.
```

## Usage

### Run the application

To run the application, run this command:

```bash
COMMIT_SHA={commit_sha} PROW_CONFIG_FILE={prow_config_file} PROW_JOBS_DIRECTORY={prow_jobs_directory} go run main.go
```

### Environment variables

Use the following environment variables to configure the application:

| Name                      | Required  | Default                   | Description |
|---------------------------|-----------|---------------------------|-------------|

> | **INITIAL_SLEEP_TIME**    | No        | `1m`                      | The initial sleep time for the application |

Reviewer: Please capitalize the values in the `Required` field.

| **RETRY_INTERVAL**        | No        | `15s`                     | The interval between re-fetching statuses |
| **TIMEOUT**               | No        | `15m`                     | The timeout of waiting for successful jobs |
| **API_ORIGIN**            | No        | `https://api.github.com`  | The origin of the GitHub API |

> | **REPO_OWNER**            | No        | `kyma-project`            | Username or organization name, that owns the repository |

Reviewer: ```suggestion
| **REPO_OWNER**            | No        | `kyma-project`            | The username or organization name of the repository owner |
```

| **REPO_NAME**             | No        | `kyma`                    | The name of the repository |
| **COMMIT_SHA**            | Yes       |                           | The commit SHA |
| **GITHUB_TOKEN**          | No        |                           | The authorization token for GitHub API |

> | **JOB_NAME_PATTERN**      | No        | `components`              | The Regexp to filter dependant statuses |

Reviewer: ```suggestion
| **JOB_NAME_PATTERN**      | No        | `components`              | The regexp to filter dependant statuses |
```

# Job Guard 

## Overview

Job Guard is a simple tool that fetches all statuses for GitHub pull requests and waits for some to finish.

The main purpose of the Job Guard is to delay running integration jobs, which depend on components ones. The utility is run as a guard for integration tests.

## Usage

### Run the application

To run the application, run this command:

```bash
PULL_NUMBER={pullNumber} go run main.go
```

Replace values in curly braces with proper details, where:
- `{pullNumber}` is the pull request number.

The service listens on port `3000`.

### Environmental variables

Use the following environment variables to configure the application:

| Name | Required | Default | Description |
|------|----------|---------|-------------|
| **PULL_NUMBER** | Yes | | The pull request number |
| **GITHUB_TOKEN** | No | | The authorization token for GitHub API|

> | **JOB_FILTER_SUBSTRING** | No | `components` | The substring that only dependant job contains in the status name |

Reviewer 2: to be more generic we should allow selecting job names by providing regexp
